### PR TITLE
Added tests for Test Connection

### PR DESCRIPTION
**Pull request by dchristian3188**
First pass of test for Test-Connection

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,37 @@
+Describe "Test-Connection" -Tags "CI" {
+    if ( -not($IsWindows)) {
+         $PSDefaultParameterValues["it:skip"] = $true
+    }
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: Please use the pattern https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/WritingPesterTests.md#skipping-tests-in-bulk

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    Context "Querying localhost" {
+        it "Gets the right IP Address" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: Please use a capital letter "It".
Below too.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            $localHost = Test-Connection -ComputerName localhost -Count 1
+            $localHost.IPV4Address | should be '127.0.0.1'
+            $localhost.IPV6Address | should be '::1'
+        }
+
+        $countCases = @(
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: Please move all initializations to `BeforeAll` block.
Below too.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        it "Respects the quiet parameter on a <message> computer" -TestCases $quietTests {
+            param($computername,$result)
+            Test-Connection -ComputerName $computername -Count 1 -Quiet | should be $result
+        }
+
+        it "Respects the timeout parameter" {
+            $start = [datetime]::now
+            Test-Connection -Computer _bad_computer_name_ -Timeout 10000 -ea silentlycontinue
+            $end = [datetime]::now
+            ($end - $start).TotalSeconds | should BeGreaterThan 10.0
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: I believe we can use a lower value - 4 sec?
We should add `-Count 1`.
Please use capital letter `Should ` and `Should Be` above too.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,37 @@
+Describe "Test-Connection" -Tags "CI" {
+    if ( -not($IsWindows)) {
+         $PSDefaultParameterValues["it:skip"] = $true
+    }
+    Context "Querying localhost" {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: I don't see any reason to use `Context` here.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+            $localHost = Test-Connection -ComputerName localhost -Count 1
+            $localHost.IPV4Address | should be '127.0.0.1'
+            $localhost.IPV6Address | should be '::1'
+        }
+
+        $countCases = @(
+            @{count = 2}
+            @{count = 3}
+        )
+        it "Respects the count parameter counting to <count>" -TestCases $countCases {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: I believe we can remove "Respects" from all titles.

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+            param($count)
+            (Test-Connection -ComputerName localhost -Count $count).Count | should be $count
+        }
+        
+        $quietTests = @(
+            @{computerName = 'localhost'; message = 'good' ; result = $true}
+            @{computerName = '_fake_computer_namex'; message = 'bad' ; result = $false}
+        )
+        it "Respects the quiet parameter on a <message> computer" -TestCases $quietTests {
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: The title and "good"/"bad" do not reflect the purpose of the test.

**Review by iSazonov — COMMENTED** [review]
@dchristian3188 Thanks for your contribution!

**Comment by dchristian3188** [comment]
thanks for the feedback @iSazonov. I'll get started on the changes

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+            param($computername,$result)
+            Test-Connection -ComputerName $computername -Count 1 -Quiet | should be $result
+        }
+
+        it "Respects the timeout parameter" {
+            $start = [datetime]::now
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: I'm concerned that this test takes so long - should it be moved to it's own describe block and marked as `slow`?
also, could it be written as?
`(measure-command { test-connection -computer _bad_computer_name_ -timeout 10000 -ea silentlycontinue}).TotalSeconds | should begreaterthan 10.0`

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+    AfterAll {
+        $global:PSDefaultParameterValues = $originalDefaultParameterValues
+    }
+    It "Uses the timeout parameter" {
+        (Measure-Command {
+                Test-Connection -Computer _bad_computer_name_ -Timeout 10000 -ea silentlycontinue
+            }).TotalSeconds | should BeGreaterThan 10.0
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: I totally discard the test. Here "TimeOut" applies to the PingStatus. Please read docs https://msdn.microsoft.com/en-us/library/aa394350(v=vs.85).aspx
and codes
https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs#L380
https://github.com/PowerShell/PowerShell/blob/master/src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs#L410

File: test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1 [hunk 10]
```diff
@@ -23,35 +23,17 @@ Describe "Test-Connection" -Tags "CI" {
...
-    AfterAll {
-        $global:PSDefaultParameterValues = $originalDefaultParameterValues
-    }
-    It "Uses the timeout parameter" {
-        (Measure-Command {
-                Test-Connection -Computer _bad_computer_name_ -Timeout 10000 -ea silentlycontinue
-            }).TotalSeconds | should BeGreaterThan 10.0
-    }
-}
+}
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1]: Please add Newline at EOF.

**Comment by dchristian3188** [comment]
changes added, I'll get this right eventually :)

**Comment by iSazonov** [comment]
LGTM.

@dchristian3188 Thanks for your contribution!

**Comment by daxian-dbw** [comment]
@dchristian3188 Thanks for the contribution!

**Pull request merged**